models/build_request.go

```diff
@@ -1,22 +1,98 @@
 package models
 
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
+
 // BuildRequest struct
+// TODO: Convert the return sms and wait return sms to boolean
 type BuildRequest struct {
-	SMSID         string `json:"smsid"`
-	SMSTitle      string `json:"smstitle"`
-	SMSText       string `json:"smstext"`
-	MobileNumber  string `json:"mobilenumber"`
-	OriginalURL   string `json:"originalurl"`
-	OriginalURL2  string `json:"originalurl2"`
-	OriginalURL3  string `json:"originalurl3"`
-	OriginalURL4  string `json:"originalurl4"`
-	Status        string `json:"status"`
-	ReturnSMS     string `json:"returnsms"`
-	WaitReturnSMS string `json:"waitreturnsms"`
-	Type          string `json:"type"`
-	Au            string `json:"au"`
-	Docomo        string `json:"docomo"`
-	Softbank      string `json:"softbank"`
-	Gateway       string `json:"gateway"`
-	Rakuten       string `json:"rakuten"`
+	SMSID         string
+	SMSTitle      string
+	SMSText       string
+	MobileNumber  string
+	OriginalURL   string
+	OriginalURL2  string
+	OriginalURL3  string
+	OriginalURL4  string
+	Status        string
+	ReturnSMS     string
+	WaitReturnSMS string
+	Type          string
+	Au            AuConfig
+	Docomo        DocomoConfig
+	Softbank      SoftbankConfig
+	Gateway       GatewayConfig
+	Rakuten       RakutenConfig
+	Sim           SimConfig
+}
+
+// NewBuildRequest returns ...
+func NewBuildRequest() BuildRequest {
+	return BuildRequest{
+		Au:       AuConfig{NumberOfRetries: 1},
+		Docomo:   DocomoConfig{NumberOfRetries: 1, RetryInterval: []int{0}},
+		Softbank: SoftbankConfig{NumberOfRetries: 1, RetryInterval: []int{5}},
+		Gateway:  GatewayConfig{NumberOfRetries: 1, RetryInterval: []int{1}},
+		Rakuten:  RakutenConfig{NumberOfRetries: 1, RetryInterval: []int{2}},
+		Sim:      SimConfig{NumberOfRetries: 1},
+	}
+}
+
+// MarshalJSON overrides the custom serialization to prepare the data for the MediaSms API
+func (br BuildRequest) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		SMSID         string `json:"smsid,omitempty"`
```

Comment:
How do you think to write a comment like it's required if we use 2 ways sms ? It was written on media4u 2ways documentation. In our cause, we already implemented in `mediasms.go` though.

Reply:
I added validation step